src/CredentialsManagement.h

```diff
@@ -19,63 +19,147 @@
 #ifndef CREDENTIALSMANAGEMENT_H
 #define CREDENTIALSMANAGEMENT_H
 
+// Qt
 #include <QtWidgets>
+
+// Application
 #include "WSClient.h"
 #include "CredentialsModel.h"
 
 namespace Ui {
 class CredentialsManagement;
 }
+class CredentialModel;
+class CredentialModelFilter;
 
 class CredentialsManagement : public QWidget
 {
     Q_OBJECT
 
-public:
+public:    
+    //-------------------------------------------------------------------------------------------------
+    // Constructor & destructor
+    //-------------------------------------------------------------------------------------------------
+
+    //! Constructor
     explicit CredentialsManagement(QWidget *parent = 0);
+
+    //! Destructor
     ~CredentialsManagement();
 
+    //! Set WS client
     void setWsClient(WSClient *c);
 
-signals:
-    void wantEnterMemMode();
-    void wantSaveMemMode();
-
 public slots:
-    bool confirmDiscardUneditedCredentialChanges(QModelIndex idx = {});
+    //-------------------------------------------------------------------------------------------------
+    // Public slots
+    //-------------------------------------------------------------------------------------------------
+
+    //! Confirm discard unedited credential changes
+    bool confirmDiscardUneditedCredentialChanges(const QModelIndex &proxyIndex = {});
 
 private slots:
+    //-------------------------------------------------------------------------------------------------
+    // Private slots
+    //-------------------------------------------------------------------------------------------------
+
+    //! Enable credentials management
     void enableCredentialsManagement(bool);
+
+    //! Update quick add credentials button state
     void updateQuickAddCredentialsButtonState();
-    void updateSaveDiscardState(QModelIndex idx = {});
 
-    void onPasswordUnlocked(const QString & service, const QString & login, const QString & password, bool success);
-    void onCredentialUpdated(const QString & service, const QString & login, const QString & description, bool success);
+    //! Update save discard state
+    void updateSaveDiscardState(const QModelIndex &proxyIndex=QModelIndex());
+
+    //! Password unlocked
+    void onPasswordUnlocked(const QString &service, const QString &login, const QString &password, bool success);
 
-    void saveSelectedCredential(QModelIndex idx = {});
+    //! Credential updated
+    void onCredentialUpdated(const QString &service, const QString &login, const QString &description, bool success);
 
+    //! Save selected credential
+    void saveSelectedCredential(const QModelIndex &proxyIndex=QModelIndex());
+
+    //! Button EnterMMM clicked
     void on_pushButtonEnterMMM_clicked();
+
+    //! Button discard clicked
```

Comment:
Why all those useless comments?
Please avoid that, it helps no one, it's juste noise.
https://testing.googleblog.com/2017/07/code-health-to-comment-or-not-to-comment.html